integrasjonspunkt/src/main/java/no/difi/meldingsutveksling/nextmove/message/FileMessagePersister.java

```diff
@@ -26,44 +26,78 @@ public FileMessagePersister(IntegrasjonspunktProperties props) {
     }
 
     @Override
-    public void write(ConversationResource cr, String filename, byte[] message) throws IOException {
-        String filedir = getConversationFiledirPath(cr);
-        File localFile = new File(filedir+filename);
+    public void write(String messageId, String filename, byte[] message) throws IOException {
+        String filedir = getMessageFiledirPath(messageId);
+        File localFile = new File(filedir + filename);
         localFile.getParentFile().mkdirs();
 
-        if (props.getNextmove().getApplyZipHeaderPatch() && props.getNextmove().getAsicfile().equals(filename)){
-            BugFix610.applyPatch(message, cr.getConversationId());
+        if (props.getNextmove().getApplyZipHeaderPatch() && ASIC_FILE.equals(filename)) {
+            BugFix610.applyPatch(message, messageId);
         }
 
         try (FileOutputStream os = new FileOutputStream(localFile);
              BufferedOutputStream bos = new BufferedOutputStream(os)) {
             bos.write(message);
-            bos.flush();
-            bos.close();
         } catch (IOException e) {
-            log.error("Could not write asic container to disc.", e);
+            log.error("Could not write asic container to disk.", e);
             throw e;
         }
     }
 
     @Override
-    public byte[] read(ConversationResource cr, String filename) throws IOException {
-        String filedir = getConversationFiledirPath(cr);
-        File file = new File(filedir+filename);
+    public void writeStream(String messageId, String filename, InputStream inputStream, long size) throws IOException {
+        String filedir = getMessageFiledirPath(messageId);
+        File localFile = new File(filedir + filename);
+        localFile.getParentFile().mkdirs();
+
+        try (FileOutputStream os = new FileOutputStream(localFile);
+             BufferedOutputStream bos = new BufferedOutputStream(os)) {
+            int bytes = IOUtils.copy(inputStream, bos);
+            log.debug("Storing {}: {}", filename, humanReadableByteCount(bytes));
+        } catch (IOException e) {
+            log.error("Could not write asic container to disk.", e);
+            throw e;
+        }
+    }
+
+    private String humanReadableByteCount(int bytes) {
+        int unit = 1000;
+        if (bytes < unit) return bytes + " B";
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') At most one statement is allowed per line, but 2 statements were found on this line. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AS00122)